docs/docs/sourcing-from-etsy.md

```diff
@@ -0,0 +1,113 @@
+
+## Using the Etsy source plugin
+
+You can use the `gatsby-source-etsy` plugin to pull featured listings and images from your Etsy shop into your Gatsby site. To install it, run:
+
+```bash
+npm install gatsby-source-etsy
+```
+
+### Configuring the plugin
+
+You will then need to add the plugin, your API key, and your shop ID to your `gatsby-config.js` file. That said, finding your shop ID can be confusing. While your shop does have an ID _number_, you don't actually need this to make requests against the Etsy API. You can use your shop's _name_ as a unique identifier if you don't know your shop ID.
+
+```js:title=gatsby-config.js
+plugins: [
+  {
+    resolve: "gatsby-source-etsy",
+    options: {
+      // highlight-start
+      apiKey: "your api key here",
+      shopId: "your shop id or shop name here",
+      // highlight-end
+      language: "en", // optional
+    },
+  },
+],
+```
+
+### Testing your queries
+
+Once you have everything set up, run `gatsby develop` to start your site locally. At that point, you should be able to query for data related to your featured listings in [GraphiQL](/docs/running-queries-with-graphiql/). As an example, the following query gets the total number of featured listings in the shop as well as the price, title, and description of each item.
+
+```graphql
+allFeaturedEtsyListing {
+  totalCount
+    nodes {
+      price
+      title
+      description
+    }
+  }
+```
+
+Make sure you have _featured_ items! To edit your store, go to the Shop Manager and select your Etsy shop under "Sales Channels". You should see an option to edit or enable featured items.
+
+## Displaying Etsy listings
+
+You can pull your Etsy listing data into a page or component by composing a GraphQL query and specifying the data you would like to display. For example, consider the following query, which gets the title, price, URL, and an image of each featured listing.
+
+```graphql
+allFeaturedEtsyListing {
+  nodes {
+    title
+    price
+    url
+    description
+    childEtsyListingImage {
+      childFile {
+        childImageSharp {
+          fixed(width: 400, height: 400) {
+            ...GatsbyImageSharpFixed
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Using this query on a page or in a component will allow you to display your listings. The following example loops through the featured items and displays them within a component.
+
+```js
+{
+  data.allFeaturedEtsyListing.nodes.map(item => (
+    <>
+      <a href={item.url}>
+        <h2>{item.title}</h2>
+      </a>
+      <Img fixed={item.childEtsyListingImage.childFile.childImageSharp.fixed} />
```

Comment:
Is there a way to add variable alt text here?

Reply:
So there's not a _great_ solution for this. Etsy actually doesn't give shop owners a place to add alt text. (Another reason to pull your listings into your own site!) The best I could come up with to do this with the data provided is repeating the title of the item.

I was thinking the alt text was taken from the filename by default but I may have been thinking of MDX... As it is, the alt attribute is still added but it's empty, which means the images should be ignored. Given that the image is sandwiched between the item's name and description, I'm not sure if it makes sense to repeat that information or leave it as is. Tagging @madalynrose!

Reply:
Either way, I think it's worth noting in the text.

Reply:
+1 what Aisha said. if the alt text is going to be repeating information already present, that can get really annoying with a screen reader. keeping the image purely presentational makes sense unless we can communicate extra information about it (e.g. color variation)

Reply:
Makes sense! Hadn't thought about the images being sandwiched between that info. Thanks for thinking it through so thoroughly :)